smart_home/src/services/temperature_service.cc

```diff
@@ -0,0 +1,66 @@
+#include <iostream>
+#include "../../include/services/temperature_service.h"
+#include "../../include/sensors/temperature_sensor.h"
+#include "../../include/devices/ac_unit.h"
+
+namespace services {
+
+void TemperatureService::AddSensor(sensors::Sensor* sensor) {
+    sensors_.push_back(sensor);
```

Comment:
although the code is correct from compiler POV, you should make sure the sensor you're adding is a temperature sensor. This would improve your design of your class by making sure no other kind of sensor could be added by mistake